platforms/core-configuration/declarative-dsl-provider/src/integTest/groovy/org/gradle/internal/declarativedsl/agp/DeclarativeAgpSmokeSpec.groovy

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.declarativedsl.agp
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.internal.jvm.Jvm
+import org.gradle.test.fixtures.dsl.GradleDsl
+import org.junit.Assume
+
+class DeclarativeAgpSmokeSpec extends AbstractIntegrationSpec {
+
+    /**
+     * TODO: no Android version published to the google() repo has the `com.android.ecosystem` plugin yet.
+     *   Once there is an AGP version with DCL support published there, remove the hardcoded Maven repo URLs and use {@link org.gradle.integtests.fixtures.versions.AndroidGradlePluginVersions}.
+     */
```

Comment:
This can be addressed as soon as this is merged
* https://github.com/gradle/gradle/pull/32515